presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -1738,6 +1740,41 @@ public void dropView(ConnectorSession session, SchemaTableName viewName)
         return tableNames.build();
     }
 
+    @Override
+    public Map<String, Object> getSchemaProperties(ConnectorSession session, CatalogSchemaName schemaName)
+    {
+        if (!filterSchema(schemaName.getSchemaName())) {
+            // "information_schema" and "sys" have no properties
```

Comment:
Please do not use `filterSchema` but just make sure that requested schema is one of these. And if it is the case I think we should throw an exception.

@findepi I think you are more familiar with `sys` schema. Are ok with my suggestion?

Reply:
That's precisely what `HiveMetadata.filterSchema` does, though. It's a static method that checks for these schemas (I suppose it's badly named).
I can throw an exception instead, but I'd just do the same check with extra code.

Reply:
Should I add a new error code to `HiveErrorCode`?

Reply:
Have look. Throwing NOT_SUPPORTED exception now.

Reply:
Can engine call `getSchemaProperties` without calling `schemaExists` first? (Does engine guarantee that?)

* yes => sys needs to be filtered out; and we should return NOT_FOUND (because we pretend the "sys" schema does not exist, as indicated by `false` from `schemaExists`)
* no => `checkState` would be more appropriate: `checkState(filterSchema(..), "Schema is not accessible: %s", schemaName)`

Reply:
This is indeed only called from `ShowQueriesRewrite,Visitor.visitShowCreate(...)`, which does check `schemaExists` first and throws SCHEMA_NOT_FOUND from there. (Although it's not guaranteed that nobody will add such a call in the future.)

Added checkState bit...